File: 14-week-11/02-days-4-to-5--breadth-first-graph-traversal/ruby/find_distance.rb
def find_distance(graph, vertex_a, vertex_b)
  # type your code in here
end

if __FILE__ == $PROGRAM_NAME
  graph = {
    jan: [:john, :jambaby],
    john: [:carl],
    jambaby: [],
    carl: [:jambaby],
    dave: []
  }

  puts "Expecting: 2"
  puts find_distance(graph, :jan, :carl)

  puts

  puts "Expecting: -1"
  puts find_distance(graph, :dave, :carl)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: 14-week-11/02-days-4-to-5--breadth-first-graph-traversal/javascript/find_distance.js
function findDistance(graph, vertexA, vertexB) {
  // type your code here
}

if (require.main === module) {
  // add your own tests in here
  const graph = {
    jan: ["john", "jambaby"],
    john: ["carl"],
    jambaby: [],
    carl: ["jambaby"],
    dave: []
  };

  console.log("Expecting: 2");
  console.log(findDistance(graph, "jan", "carl"));

  console.log("");

  console.log("Expecting: -1");
  console.log(findDistance(graph, "dave", "carl"));
}

module.exports = findDistance;

// Please add your pseudocode to this file
// And a written explanation of your solution
